From 7273820709675377a0cf8a1445f92007e65fa008 Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Tue, 20 Aug 2013 15:09:51 +0200
Subject: [PATCH 072/212] pc: SandyBridge rhel6.x compat fixes

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1377011392-9336-7-git-send-email-ehabkost@redhat.com>
Patchwork-id: 53614
O-Subject: [RHEL7 PATCH 6/7] pc: SandyBridge rhel6.x compat fixes
Bugzilla: 918907
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Bugzilla: 918907
Upstream status: not applicable

This patch implements the following changes, to match the SandyBridge
CPU features on RHEL-6:

 * rhel6.3.0 and older don't have tsc-deadline on SandyBridge
   (commit 10ce057)
 * rhel6.5.0 and older don't have rdtscp on SandyBridge

Signed-off-by: Eduardo Habkost <ehabkost@raisama.net>
---
 hw/i386/pc_piix.c | 4 ++++
 1 file changed, 4 insertions(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/pc_piix.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index b3dcc21..6a1cc45 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1038,6 +1038,8 @@ static void pc_init_rhel650(MachineState *machine)
              CPUID_EXT2_CX8 | CPUID_EXT2_MCE | CPUID_EXT2_PAE | CPUID_EXT2_MSR |
              CPUID_EXT2_TSC | CPUID_EXT2_PSE | CPUID_EXT2_DE | CPUID_EXT2_FPU,
              0);
+    x86_cpu_compat_set_features("SandyBridge", FEAT_8000_0001_EDX,
+                                0, CPUID_EXT2_RDTSCP);
     pc_init_rhel700(machine);
 }
 
@@ -1123,6 +1125,8 @@ static void pc_init_rhel630(MachineState *machine)
 {
     x86_cpu_compat_disable_kvm_features(FEAT_KVM, KVM_FEATURE_PV_EOI);
     enable_compat_apic_id_mode();
+    x86_cpu_compat_set_features("SandyBridge", FEAT_1_ECX,
+                                0, CPUID_EXT_TSC_DEADLINE_TIMER);
     pc_init_rhel640(machine);
 }
 
-- 
1.7.1

